sys/usb/usbus/dfu/dfu.c

```diff
@@ -112,6 +112,12 @@ void usbus_dfu_init(usbus_t *usbus, usbus_dfu_device_t *handler, unsigned mode)
                 USB_DFU_STATE_DFU_IDLE : USB_DFU_STATE_APP_IDLE;
 
     usbus_register_event_handler(usbus, (usbus_handler_t *)handler);
+#ifdef MODULE_RIOTBOOT_USB_DFU
```

Comment:
This looks like a mayor foot-gun to, as this among the last places I would expect a `ztimer_init()` call. Would moving that call prior to `riotboot_usb_dfu_init(0);` in line 67/68 of `bootloaders/riotboot_dfu/main.c` solve the issue as well?

Reply:
I tried this week-end. But I ran into another weird issue:
Flashing `examples/blinky` works with the patch you mentioned but the application doesn't start. I suspect an issue with a double init (initialized by riotboot then by auto_init when the slot boots). I didn't dig deeper but I can eventually have a closer look at this in the next days.

Reply:
Another option would be to move `ztimer_init` within `riotboot_usb_dfu_init()` in `sys/riotboot/usb_dfu.c`